spec/unit/shoulda/matchers/active_record/validate_uniqueness_of_matcher_spec.rb

```diff
@@ -726,6 +726,51 @@
     end
   end
 
+  context 'when the writer method for the attribute changes incoming values' do
+    context 'and ignoring_case_sensitivity is not specified' do
+      it 'raises a CouldNotSetAttributeError' do
+        model = define_model_validating_uniqueness(
+          attribute_name: :name,
+          validation_options: { case_sensitive: false },
+        )
+
+        model.class_eval do
+          def name=(name)
+            super(name.upcase)
+          end
+        end
+
+        assertion = lambda do
+          expect(model.new).
+            to validate_uniqueness_of(:name).
+            case_insensitive
+        end
+
+        expect(&assertion).to raise_error(
+          Shoulda::Matchers::ActiveModel::AllowValueMatcher::CouldNotSetAttributeError
+        )
+      end
+    end
+
+    context 'and ignoring_case_sensitivity is specified' do
+      it 'accepts (and not raise an error)' do
+        model = define_model_validating_uniqueness(
+          attribute_name: :name,
```

Comment:
I'm noticing that in the first test, you have the validation qualified with `case_sensitive: false`, but you don't have that here.

I'm kind of forgetting what the original bug was here. Is that something we need to have here, you think? Or, should we have two sets of tests, one where the validation is  `case_sensitive: true`, one where it's  `case_sensitive: false`?